extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -347,22 +396,25 @@ private static void extractClientVersionAndKey() throws IOException, ExtractionE
 
         try {
             key = Parser.matchGroup1("INNERTUBE_API_KEY\":\"([0-9a-zA-Z_-]+?)\"", html);
-        } catch (Parser.RegexException e) {
+        } catch (final Parser.RegexException e) {
             try {
                 key = Parser.matchGroup1("innertubeApiKey\":\"([0-9a-zA-Z_-]+?)\"", html);
-            } catch (Parser.RegexException ignored) {
+            } catch (final Parser.RegexException ignored) {
             }
         }
+        keyAndVersionExtracted = true;
```

Comment:
Shouldn't this be set to `false` if the key or the version could not be extracted? Or throw an exception? I know the exception part is handled in `getClientVersion()`, but since you set `keyAndVersionExtracted` to true, `extractClientVersionAndKey()` will never be called again.

Reply:
Done